Mocks are not working on particular devices after update Android SDK from 33 to 34

After seeing #3075, I've updated our project to Mockito 5.7.0 (btw I also tried 5.5.0), but even after that mocks are not working anymore on few on my devices. Then I was able to reproduce this issue in sample app which can be found [here](https://github.com/jukov/MockitoBugTest).

I am trying to run this test:

```
interface TestInterface {
    fun foo(): Int
}

@RunWith(AndroidJUnit4::class)
class MockitoBugTest {

    @Mock private val testInterface: TestInterface = mock(TestInterface::class.java)

    @Before
    fun setup() {
        Mockito.`when`(testInterface.foo()).thenReturn(42)
    }

    @Test
    fun foo() {
        println("Success")
    }
}
```

After what I receive the following error:

<details>
  <summary>Stacktrace</summary>

```
java.lang.ExceptionInInitializerError
at org.mockito.internal.debugging.LocationFactory$DefaultLocationFactory.create(LocationFactory.java:48)
at org.mockito.internal.debugging.LocationFactory.create(LocationFactory.java:19)
at org.mockito.internal.debugging.LocationFactory.create(LocationFactory.java:15)
at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:56)
at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:161)
at info.jukov.mockitobugtest.TestInterface$MockitoMock$KqIWwkGD.foo(Unknown Source:15)
at info.jukov.mockitobugtest.MockitoBugTest.setup(MockitoBugTest.kt:33)
... 32 trimmed
Caused by: java.lang.ClassCastException: java.util.stream.ReferencePipeline$Head cannot be cast to j$.util.stream.Stream
at org.mockito.internal.debugging.LocationImpl$$ExternalSyntheticLambda0.apply(Unknown Source:0)
at java.lang.StackStreamFactory$StackFrameTraverser.consumeFrames(StackStreamFactory.java:601)
at java.lang.StackStreamFactory$AbstractStackWalker.doStackWalk(StackStreamFactory.java:316)
at java.lang.StackStreamFactory$AbstractStackWalker.callStackWalk(StackStreamFactory.java:436)
at java.lang.StackStreamFactory$AbstractStackWalker.beginStackWalk(StackStreamFactory.java:380)
at java.lang.StackStreamFactory$AbstractStackWalker.walk(StackStreamFactory.java:251)
at java.lang.StackWalker.walk(StackWalker.java:503)
at org.mockito.internal.debugging.LocationImpl.stackWalk(LocationImpl.java:134)
at org.mockito.internal.debugging.LocationImpl.framesToSkip(LocationImpl.java:123)
at org.mockito.internal.debugging.LocationImpl.<clinit>(LocationImpl.java:59)
... 40 more
```
  
</details>

Also I've debugged mock process and found that some problem occurred in `SubclassByteBuddyMockMaker.createMock` method:

```
public <T> T createMock(MockCreationSettings<T> settings, MockHandler handler) {
        ...
        try {
            mockInstance = instantiator.newInstance(mockedProxyType);
            MockAccess mockAccess = (MockAccess) mockInstance;
            mockAccess.setMockitoInterceptor(new MockMethodInterceptor(handler, settings)); // Here

            return ensureMockIsAssignableToMockedType(settings, mockInstance);
        } catch (ClassCastException cce) {
           ...
        }
```

After executing `setMockitoInterceptor` debugger shows me following error:
```Method threw 'java.lang.NoClassDefFoundError' exception. Cannot evaluate info.jukov.mockitobugtest.TestInterface$MockitoMock$tH4GO8Yq.toString()```


**Additional details:**

**Tests are not working on:**
🚫Google Pixel 6 Pro (Android 14)
🚫[Solana Mobile Saga](https://solanamobile.com/hardware) (Android 12)
🚫Emulator Android 14

**Tests are working on:**
✅OnePlus 9 with Lineage OS (Android 13)
✅Xiaomi Redmi 10 2022 (MIUI 12.5.26, Android 11)
✅Emulator Android 13
✅Emulator Android 12
✅Emulator Android 11

**Versions:**
Android target SDK: 34
Mockito version: 5.7.0 (also tried 5.5.0)
Kotlin version: 1.9.20 (also tried 1.8.22)
Java version: 11 (also tried 17)

--- a/mockito-core/src/main/java/org/mockito/internal/debugging/LocationFactory.java
+++ b/mockito-core/src/main/java/org/mockito/internal/debugging/LocationFactory.java
@@ -4,6 +4,8 @@
  */
 package org.mockito.internal.debugging;
 
+import org.mockito.internal.util.AndroidPlatform;
+import org.mockito.internal.util.Platform;
 import org.mockito.invocation.Location;
 
 public final class LocationFactory {
@@ -24,6 +26,9 @@ public final class LocationFactory {
     }
 
     private static Factory createLocationFactory() {
+        if (Platform.isAndroid() && !AndroidPlatform.isStackWalkerUsable()) {
+            return new Java8LocationFactory();
+        }
         try {
             // On some platforms, like Android, the StackWalker APIs may not be
             // available, in this case we have to fallback to Java 8 style of stack

--- a/mockito-core/src/main/java/org/mockito/internal/util/AndroidPlatform.java
+++ b/mockito-core/src/main/java/org/mockito/internal/util/AndroidPlatform.java
@@ -1,0 +1,33 @@
+/*
+ * Copyright (c) 2025 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.util;
+
+public class AndroidPlatform {
+
+    private static int getSdkInt() {
+        try {
+            return Class.forName("android.os.Build$VERSION").getField("SDK_INT").getInt(null);
+        } catch (ReflectiveOperationException e) {
+            return 0;
+        }
+    }
+
+    private static int getExtensionVersion(int sdk) {
+        try {
+            return (int)
+                    Class.forName("android.os.ext.SdkExtensions")
+                            .getMethod("getExtensionVersion", int.class)
+                            .invoke(null, sdk);
+        } catch (ReflectiveOperationException e) {
+            return 0;
+        }
+    }
+
+    public static boolean isStackWalkerUsable() {
+        // StackWalker on Android had a bug that is fixed in Android Baklava (36)
+        // or SDK extension train M2025-05 (17) and later. See https://r.android.com/3548340.
+        return getSdkInt() >= 36 || getSdkInt() >= 31 && getExtensionVersion(31) >= 17;
+    }
+}

--- a/mockito-integration-tests/android-tests/src/androidTest/java/org/mockitousage/androidtest/AaaMustRunFirstLocationTests.kt
+++ b/mockito-integration-tests/android-tests/src/androidTest/java/org/mockitousage/androidtest/AaaMustRunFirstLocationTests.kt
@@ -1,0 +1,26 @@
+package org.mockitousage.androidtest
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.withSettings
+import org.mockito.internal.creation.proxy.ProxyMockMaker
+
+@RunWith(AndroidJUnit4::class)
+class AaaMustRunFirstLocationTests {
+
+    /**
+     *  Regression test for https://github.com/mockito/mockito/issues/3171 and
+     *  https://github.com/linkedin/dexmaker/issues/190.
+     *
+     *  Bug only triggers if the first time LocationImpl is used is with ProxyMockMaker, therefore
+     *  this test must be the first to run.
+     */
+    @Test
+    fun mockAndUseInterfaceWithProxyMockMaker() {
+        val basicInterface = mock(BasicInterface::class.java,
+            withSettings().mockMaker(ProxyMockMaker::class.java.name))
+        basicInterface.interfaceMethod()
+    }
+}

